### PR TITLE
Save/Delete data to CoreData and update status button

### PR DESCRIPTION
**Pull request by ghost**
## Related Tickets
- [#Task 73910: Handle event](https://edu-redmine.sun-asterisk.vn/issues/73910)
## WHAT
- Save data to core data
- Delete data to core data
- update status button favorite
## Evidence (Screenshot or Video)
Uploading Screen Recording 2024-03-27 at 15.07.00.mov…
## Review Checklist

Category | View Point | Description | Self review | Reviewer2 (name)
--- | --- | --- | --- | ---
Conventions | Sun* coding conventions | https://sal.vn/Zts4P8 |<li>- [ ] yes</li>|<li>- [ ] yes</li>
Message | Sun* commit message | Template: [TASK/FREATURE/BUG/SUPPORT][RedmineID] + message |<li>- [ ] yes</li>|<li>- [ ] yes</li>  
Redmine | Sun* Redmine working process  | https://sal.vn/piPoQn |<li>- [ ] yes</li>|<li>- [ ] yes</li>  

## Notes (Optional)
*(Impacted Areas in Application(List features, api, models or services that this PR will affect))*

*(List gem, library third party add new)*

*(Other notes)*

### CODE REVIEW

File: Weather_App/Weather_App/Scenes/Map/MapViewController.swift [hunk 1]
```diff
@@ -23,6 +24,8 @@ final class MapViewController: UIViewController {
     
     private let weatherRepository: WeatherRepository = WeatherAPIRepository()
     private var locationManager = LocationManager.shared
+    private var currentWeatherData: WeatherCurrent?
+    private var isFavorite: Bool = false
```
Comment (ghost) [inline Weather_App/Weather_App/Scenes/Map/MapViewController.swift]:
```suggestion
    private var isFavorite = false
```

File: Weather_App/Weather_App/Scenes/Map/MapViewController.swift [hunk 2]
```diff
@@ -103,5 +107,76 @@ extension MapViewController {
...
         }
+        updateFavoriteStatus(for: weatherCurrent.nameCity)
+    }
+}
+
+extension MapViewController {
+    @IBAction private func favoriteButtonTapped(_ sender: Any) {
+        guard let weatherData = currentWeatherData else { return }
+        if isFavorite {
+            if let weatherEntity = fetchWeatherEntity(for: weatherData.nameCity) {
```
Comment (ghost) [inline Weather_App/Weather_App/Scenes/Map/MapViewController.swift]:
```suggestion
        if isFavorite, let weatherEntity = fetchWeatherEntity(for: weatherData.nameCity) {
```

File: Weather_App/Weather_App/Scenes/Map/MapViewController.swift [hunk 3]
```diff
@@ -103,5 +107,76 @@ extension MapViewController {
...
+        let image = UIImage(named: imageName)
+        favoriteButton.setImage(image, for: .normal)
+    }
+    
+    private func updateFavoriteStatus(for cityName: String) {
+        if let weatherEntity = fetchWeatherEntity(for: cityName) {
+            isFavorite = true
+        } else {
+            isFavorite = false
+        }
```
Comment (ghost) [inline Weather_App/Weather_App/Scenes/Map/MapViewController.swift]: thử cái này coi sao e
```suggestion
        isFavorite = fetchWeatherEntity(for: cityName) != nil
```

File: Weather_App/Weather_App/Scenes/Map/MapViewController.swift [hunk 4]
```diff
@@ -103,5 +107,76 @@ extension MapViewController {
...
+        do {
+            let results = try context.fetch(fetchRequest)
+            return results.first
+        } catch {
+            print("Error fetching favorite cities: \(error)")
+            return nil
+        }
+    }
+    
+    private func saveWeatherToCoreData(latitude: Double, longitude: Double, cityName: String) {
```
Comment (dongtg-1485) [inline Weather_App/Weather_App/Scenes/Map/MapViewController.swift]: không xử lý logic trực tiếp trong View như này nhé em, tách hết thành service như location

**Review by daolq-2712 — APPROVED** [review]
OK!

**Pull request merged**